modules/andes-core/broker/src/main/java/org/wso2/andes/mqtt/MQTTopicManager.java

```diff
@@ -208,7 +208,7 @@ public void removeOrDisconnectTopicSubscription(String mqttClientChannelID, Subs
                 //Will remove the subscriber cluster wide
                 try {
                     //Will indicate the disconnection of the topic
-                    if (action == SubscriptionEvent.DISCONNECT) {
+                    if (action == SubscriptionEvent.DISCONNECT && !(subscriber.getQOSLevel() == 0 && !subscriber.isCleanSession())) {
```

Comment:
We could use (!subscriber.getQOSLevel() == 0 || subscriber.isCleanSession()) instead of what's written if it's in the main path(just to simplify things)